Crash with sensitive string manipulation functions - v14 rc1

### Terraform Version
```
Terraform v0.14.0-rc1
+ provider registry.terraform.io/hashicorp/azuread v1.0.0
+ provider registry.terraform.io/hashicorp/azurerm v2.34.0
```

### Terraform Configuration Files

```terraform
terraform {
  experiments = [provider_sensitive_attrs]
}

data "azurerm_client_config" "current" {
}

resource "azurerm_resource_group" "example" {
  name     = "my-resource-group"
  location = "West US"
}

resource "random_id" "server" {
  keepers = {
    ami_id = 1
  }

  byte_length = 8
}

resource "azurerm_key_vault" "example" {
  name                = format("%s%s", "kv", random_id.server.hex)
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  tenant_id           = data.azurerm_client_config.current.tenant_id

  sku_name = "premium"

  access_policy {
    tenant_id = data.azurerm_client_config.current.tenant_id
    object_id = data.azurerm_client_config.current.object_id

    key_permissions = [
      "create",
      "get",
    ]

    secret_permissions = [
      "set",
      "get",
      "delete",
    ]
  }
}

resource "azurerm_key_vault_secret" "example" {
  name         = "secret-sauce"
  value        = split(";EntityPath=", azurerm_servicebus_queue_authorization_rule.example.primary_connection_string).0 #tried also with "${split}"
  key_vault_id = azurerm_key_vault.example.id
}

resource "azurerm_servicebus_namespace" "example" {
  name                = "tfex-servicebus-namespace"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "Standard"
}

resource "azurerm_servicebus_queue" "example" {
  name                = "tfex_servicebus_queue"
  resource_group_name = azurerm_resource_group.example.name
  namespace_name      = azurerm_servicebus_namespace.example.name
  enable_partitioning = false
}

resource "azurerm_servicebus_queue_authorization_rule" "example" { 
  name                = "ListenMessage"
  namespace_name      = azurerm_servicebus_queue.example.namespace_name
  queue_name          = azurerm_servicebus_queue.example.name
  resource_group_name = azurerm_servicebus_queue.example.resource_group_name

  listen = true
  send   = false
  manage = false

  depends_on = [azurerm_servicebus_queue.example]
}
```
### Crash Output

panic: value is marked, so must be unmarked first

### Expected Behavior
Process and create secret in keyvault

### Actual Behavior
crash

### Steps to Reproduce
1. `terraform init`
2. `terraform apply`

### References
- #26633 
Same error with sensitive variables.

--- a/go.mod
+++ b/go.mod
@@ -64,7 +64,7 @@ require (
 	github.com/hashicorp/go-uuid v1.0.1
 	github.com/hashicorp/go-version v1.2.0
 	github.com/hashicorp/hcl v0.0.0-20170504190234-a4b07c25de5f
-	github.com/hashicorp/hcl/v2 v2.7.1
+	github.com/hashicorp/hcl/v2 v2.7.2
 	github.com/hashicorp/memberlist v0.1.0 // indirect
 	github.com/hashicorp/serf v0.0.0-20160124182025-e4ec8cc423bb // indirect
 	github.com/hashicorp/terraform-config-inspect v0.0.0-20191212124732-c6ae6269b9d7

